haystack/components/builders/dynamic_prompt_builder.py

```diff
@@ -1,140 +1,24 @@
 import logging
-from typing import Dict, Any, Optional, List, Union, Set
+from typing import Dict, Any, Optional, List, Set
 
 from jinja2 import Template, meta
 
 from haystack import component
 from haystack import default_to_dict
-from haystack.dataclasses.chat_message import ChatMessage, ChatRole
 
 logger = logging.getLogger(__name__)
 
 
 @component
 class DynamicPromptBuilder:
     """
-    DynamicPromptBuilder is designed to construct dynamic prompts by processing either a list of `ChatMessage`
-    instances or a string template. It integrates with Jinja2 templating for dynamic prompt generation.
+    DynamicPromptBuilder is designed to construct dynamic prompts for the pipeline. Users can change the prompt
+    template at runtime by providing a new template for each pipeline run invocation - if needed.
 
-    In the case of `ChatMessage` instances, DynamicPromptBuilder assumes the last user message in the list as a
-    template and renders it with resolved pipeline variables and any additional template variables provided. For a
-    string template, it applies the template variables directly to render the final prompt. This dual functionality
-    allows DynamicPromptBuilder to be versatile in handling different types of prompt sources, making it suitable for
-    both chat-based and non-chat-based prompt generation scenarios.
-
-    You can provide additional template variables directly to the pipeline `run` method. They are then merged with the
-    variables resolved from the pipeline runtime. This allows for greater flexibility and customization of the
-    generated prompts based on runtime conditions and user inputs.
-
-    The following example demonstrates how to use DynamicPromptBuilder to generate a chat prompt:
+    The following example demonstrates how to use DynamicPromptBuilder:
 
     ```python
-    from haystack.components.builders import DynamicPromptBuilder
```

Comment:
We should keep the required imports DynamicPromptBuilder and GPTGenerator, Pipeline, Document, component...